### PR TITLE
add commands for working with (background) scanner

### PR DESCRIPTION
**Pull request by icewind1991**
- `occ files_antvirus:status [-v]` to get info about files in the scan queue
- `occ files_antivirus:background-scan [-v] [-m MAX]` to manually trigger the background scan
- `occ files_antivirus:scan <path>` to manually scan a single file
- `occ files_antivirus:mark <path> <scanned|unscanned>` to mark a file as scanned/unscanned

### CODE REVIEW

File: lib/Command/BackgroundScan.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2019 Robin Appelman <robin@icewind.nl>
```
Comment (artonge) [inline lib/Command/BackgroundScan.php:5]: 2023

File: lib/Command/BackgroundScan.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+				} else {
+					$output->writeln("  <error>warning: no file found for file $fileId</error>");
+				}
+			}
+			$count++;
+		}
+
+		return $count;
+	}
+}
```
Comment (artonge) [inline lib/Command/BackgroundScan.php:107]: Seem similar in each command. Can't this be shared?

File: lib/Command/Scan.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+		$this->itemFactory = $itemFactory;
+	}
+
+	protected function configure() {
+		parent::configure();
+
+		$this
+			->setName('files_antivirus:scan')
+			->setDescription('Scan a file')
+		->addArgument('file', InputArgument::REQUIRED, "Path of the file to scan");
```
Comment (artonge) [inline lib/Command/Scan.php:58]: Indentation

File: lib/Command/BackgroundScan.php (hunk 2 above)
Comment (icewind1991) [reply lib/Command/BackgroundScan.php:107]: Those were leftover from copy-paste :see_no_evil:

**Review by artonge — APPROVED** [review]

File: lib/Command/BackgroundScan.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	private BackgroundScanner $backgroundScanner;
+	private AppConfig $appConfig;
+	private IEventDispatcher $eventDispatcher;
+
+	public function __construct(AppConfig $appConfig, BackgroundScanner $backgroundScanner, IEventDispatcher $eventDispatcher) {
+		parent::__construct();
+		$this->backgroundScanner = $backgroundScanner;
+		$this->eventDispatcher = $eventDispatcher;
+		$this->appConfig = $appConfig;
+	}
```
Comment (come-nc) [inline lib/Command/BackgroundScan.php:45]:
```suggestion
	public function __construct(
		private BackgroundScanner $backgroundScanner,
		private AppConfig $appConfig,
		private IEventDispatcher $eventDispatcher,
	) {
		parent::__construct();
	}
```

Comment (come-nc) [reply lib/Command/BackgroundScan.php:45]: Same for the other new classes

Comment (icewind1991) [reply lib/Command/BackgroundScan.php:45]: The app still needs to remain compatible with older nc/php versions

File: lib/BackgroundJob/BackgroundScanner.php [hunk 5]
```diff
@@ -271,29 +211,45 @@ public function getUnscannedFiles() {
 			->andWhere($this->getSizeLimitExpression($query))
 			->setMaxResults($this->getBatchSize() * 10);
 
-		return $query->execute();
+		$result = $query->execute();
+		while (($fileId = $result->fetchOne()) !== false) {
+			yield $fileId;
+		}
```
[the remark this answers is not in the document]
Comment (icewind1991) [reply lib/BackgroundJob/BackgroundScanner.php:217]: should be fine in this case from my understanding

**Pull request merged**